### ISSUE TITLE
filtering terms for alphanumeric filter

### ISSUE THREAD

**Issue by gsfk** [ga4gh-beacon/beacon-v2#79]
I use  "alphanumeric" filters for key/operator/value queries... motivated partly by using phenopackets, where many fields (eg sex) expect values from an enum, not ontology terms. 

From what I understand, the goal of the `/filtering_terms` endpoint is to make the data in a particular beacon discoverable, but it's not clear to me what to return in /filtering_terms for these, if in fact I can return anything at all, since the only fields in a [filtering terms result](https://github.com/ga4gh-beacon/beacon-v2/blob/main/framework/src/responses/sections/beaconFilteringTermsResults.yaml) are type / id / label.. it seems odd to pack values into the "label" field. 

For a simplified example, phenopacket sex is an enum of UNKNOWN_SEX, FEMALE, MALE, OTHER_SEX. So do I want four filtering terms? eg: 

```
    {
        "type": "alphanumeric",
        "id": "subject.sex",
        "label": "UNKNOWN_SEX"
    }
    {
        "type": "alphanumeric",
        "id": "subject.sex",
        "label": "FEMALE"
    }
```
            
... and so on for MALE and OTHER_SEX. "label" doesn't really make sense in this context. I would prefer something along the lines of: 

            "type": "alphanumeric",
            "id": "subject.sex",
            "options": ["UNKNOWN_SEX", "FEMALE", "MALE", "OTHER_SEX"]

... but this seems far from the spec. Possibly the issue is that these kinds of metadata queries are not considered "terms" so weren't really expected here. But I'm puzzled why the spec for filters and the spec for filtering term results are so far apart.

**Comment by gsfk** [comment]
It would be great if I could map every key-value-operator tuple to an ontology term... that's possible for this simple case but prohibitive or outright impossible elsewhere.

**Comment by mbaudis** [comment]
@gsfk You're right w/ the numerical etc. values; I don't think we have an expression for a response element here. This would actually make sense to define this key-value-operator concept in a filtering terms response - or provide the values as a range.

But for "mappable" enums etc. -> again, mostly doable though I strongly suggest using CURIEs & resolving locally. 

Our example:

```
  "responseSummary": {
    "exists": true,
    "numTotalResults": 7980
  }
```
... filtering terms which we use e.g. for front end autocompletes in <http://progenetix.org/search/>:

* <http://progenetix.org/beacon/filteringTerms/>

Sure, this could become unmanageable... But your example of having the options listed is just slightly more compact (and we have e.g. 800 active NCIT diagnostic terms - long list).

**Comment by mbaudis** [comment]
@gsfk ... for numerical values you may want to define sensible classes (e.g.

```
- ageAtCollection:
    id: ">=18"
- ageAtCollection:
    id: ">=65"
```
... but those also exist as ontology classes, somewhere, so for the sake of cross-Beacon/aggregator compatibility should be provided as terms. See also the Phenopackets `timeElement` - obviously, for storage/transfer you may want to keep exact values _but_ for querying purposes the classes are more compatible (in principle, if implemented).

**Comment by mbaudis** [comment]
@gsfk I have tried to make the documentation a bit clearer for age etc. ... http://docs.genomebeacons.org/filters/#pseudo-numerical-value-queries 
This is based on our own way to handle this now. Still, have to think about the best representation in the filtering_terms endpoint (not doing this right now).

**Comment by jrambla** [comment]
I would suggest another approach :
```
- type: custom
  id: UNKNOWN_SEX
  label: unknown genotypic sex
  scope: subject.sex
```

as you are not actually doing any comparison here, just using a non-ontologized dictionary.

@gsfk could you elaborate on why this construct doesn't fit you use case?

**Comment by gsfk** [comment]
Is the suggestion to use scope to pick out a phenopackets path? That's interesting, although doesn't fit the current spec.

**Comment by mbaudis** [comment]
@gsfk 

> Fundamentally all I really wanted was better alignment between filteringTerms and beaconFilteringTermsResults, because I don't understand why they're so far apart.

Tru dat (and on the radar).

**Comment by gsfk** [comment]
Suggest closing: some ongoing discussion, but issue itself is resolved by #160.
